mpqp/execution/simulated_devices.py

```diff
@@ -0,0 +1,75 @@
+"""To execute a circuit on a noisy simulator that reproduces the noise model of a machine,
+one can use a :class:`SimulatedDevice`. Inheriting from :class:`~mpqp.execution.devices.AvailableDevice`
+"""
+
+# TODO: finish doc
+
+
+from typing import TYPE_CHECKING
+
+from typeguard import typechecked
+
+from mpqp.execution import AvailableDevice
+
+if TYPE_CHECKING:
+    from qiskit_aer.backends.aer_simulator import AerSimulator
+    from qiskit_aer.noise import NoiseModel as Qiskit_NoiseModel
+    from qiskit_ibm_runtime.fake_provider.fake_backend import FakeBackendV2
+
+
+class SimulatedDevice(AvailableDevice):
```

Comment:
add `ABC` as a base ? add `@abstractmethod` to the methods ?

Reply:
Actually this class is not really abstract indeed, it is more a regular class that implement all methods that are common to any simulatedDevice (IBM, but qlm and so one in the future), so we moved it to a regular class.

In the future, when we will have IBMSimulatedDevice and QLMSimulatedDevice for instance, it may be in the future that we need to add abstract functions if they have some in common, but for the moment, the added methods in IBMSimulatedDevice are quite specific